core/src/main/scala/ai/h2o/sparkling/H2OFrame.scala

```diff
@@ -47,37 +47,51 @@ class H2OFrame private(val frameId: String, val columns: Array[H2OColumn], priva
   }
 
   def convertColumnsToCategorical(columns: Array[String]): H2OFrame = {
-    val endpoint = getClusterEndpoint(conf)
     val indices = columnNames.zipWithIndex.toMap
     val selectedIndices = columns.map { name =>
       indices.getOrElse(name, throw new IllegalArgumentException(s"Column $name does not exist in the frame $frameId"))
     }
-    if (selectedIndices.isEmpty) {
+    convertColumnsToCategorical(selectedIndices)
+  }
+
+  def convertColumnsToCategorical(columnIndices: Array[Int]): H2OFrame = {
+    val nonExisting = columnNames.indices.intersect(columnIndices)
+    if (nonExisting.nonEmpty) {
+      throw new IllegalArgumentException(s"Columns with indices ${nonExisting.mkString} are not in the frame $frameId. The frame has ${columnNames.length} columns.")
+    }
+    val endpoint = getClusterEndpoint(conf)
+    if (columnIndices.isEmpty) {
       this
     } else {
       val params = Map(
-        "ast" -> MessageFormat.format(s"( assign {0} (:= {0} (as.factor (cols {0} {1})) {1} []))", frameId, stringifyArray(selectedIndices))
+        "ast" -> MessageFormat.format(s"( assign {0} (:= {0} (as.factor (cols {0} {1})) {1} []))", frameId, stringifyArray(columnIndices))
       )
       val rapidsFrameV3 = update[RapidsFrameV3](endpoint, "99/Rapids", conf, params)
       H2OFrame(rapidsFrameV3.key.name)
     }
   }
 
-  def splitToTrainAndValidationFrames(splitRatio: Double): Array[H2OFrame] = {
-    if (splitRatio >= 1.0) {
-      throw new IllegalArgumentException("Split ratio must be lower than 1.0")
+  def split(splitRatios: Array[Double]): Array[H2OFrame] = {
```

Comment:
I think it is better to keep the name just to `split`. We can do this change without any deprecation as this API has not been announced so far